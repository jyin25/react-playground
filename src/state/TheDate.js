import React from 'react';

class TheDate extends React.Component {
    constructor(props) {
        super(props)
        this.state={
            datetime: new Date()
        }
    }
  render() {
    return <p> {this.state.datetime.toLocaleString()} </p>
  }
}

export default TheDate