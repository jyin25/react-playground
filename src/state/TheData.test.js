import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import TheDate from './TheDate'

describe ('TheDate component', () => {
    it('render without crashing', () => {

        const div = document.createElement('div');

        ReactDOM.render(<TheDate />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it.skip('renders the UI as expected', () => {
        expect (
            renderer.create(<TheDate />).toJSON()
        ).toMatchSnapshot()
    })
})