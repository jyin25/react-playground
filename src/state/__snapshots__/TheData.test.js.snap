// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheDate component renders the UI as expected 1`] = `
<p>
   
  6/11/2019, 8:30:18 AM
   
</p>
`;
